### PR TITLE
New Gin modifications

### PR DESCRIPTION
**Pull request by cgars**

### CODE REVIEW

File: web/web.go [hunk 1]
```diff
@@ -48,9 +48,11 @@ func (cl *Client) Get(address string) (*http.Response, error) {
 	if err != nil {
 		return nil, err
 	}
+	req.Header.Set("content-type", "application/jsonAuthorization")
+	util.LogWrite("Performing GET with token: %s", cl.Token)
 	if cl.Token != "" {
-		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", cl.Token))
-		util.LogWrite("Added bearer token to GET")
+		req.Header.Set("Authorization", fmt.Sprintf("token %s", cl.Token))
+		util.LogWrite("Added token token to GET")
```
Comment (achilleas-k) [inline web/web.go]: `token token`

File: web/web.go [hunk 2]
```diff
@@ -68,10 +70,10 @@ func (cl *Client) Post(address string, data interface{}) (*http.Response, error)
 	if err != nil {
 		return nil, err
 	}
-
+	req.Header.Set("content-type", "application/jsonAuthorization")
 	if cl.Token != "" {
-		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", cl.Token))
-		util.LogWrite("Added bearer token to POST")
+		req.Header.Set("Authorization", fmt.Sprintf("token %s", cl.Token))
+		util.LogWrite("Added token token to POST")
```
Comment (achilleas-k) [inline web/web.go]: `token token`

File: auth/auth.go [hunk 3]
```diff
@@ -5,13 +5,32 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/url"
-	"strings"
 
 	"github.com/G-Node/gin-cli/util"
 	"github.com/G-Node/gin-cli/web"
 	"github.com/G-Node/gin-core/gin"
+	"net/http"
```
Comment (achilleas-k) [inline auth/auth.go]: This doesn't seem right. `net/http` related things should be handled in the lower level subpackage `web`. More on this in the general comment.

Comment (cgars) [inline auth/auth.go]: Well as you check for http response codes (like ==200) in the code it could be argued that it kind of belongs here.

Comment (achilleas-k) [inline auth/auth.go]: Right, in that case yeah, using it for codes is much nicer than checking against a literal.

**Comment by cgars** [comment]
That would probably require to modify web such that you can set the headers of the requests. I am all in for that, however, as this is currently not really implemented, it has been decided to patch login such  as it is.  I guess the necessary changes  this could be part o the consolidation that is about to come (switching structs etc.)

**Pull request merged**